src/app/data-model-provider/MetadataTypes.h

```diff
@@ -22,6 +22,7 @@
 #include <access/Privilege.h>
 #include <lib/core/DataModelTypes.h>
 #include <lib/support/BitFlags.h>
+#include <assert.h>
```

Comment:
I think we may be able to remove all asserts - we seem to only assert on kNoPrivilege and that is a valid value. static_assert for everything else.